sqldelight-gradle-plugin/src/test/fixtures/function-type-tests/expected/com/test/UserModel.java

```diff
@@ -0,0 +1,320 @@
+package com.test;
+
+import android.database.Cursor;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import com.squareup.sqldelight.RowMapper;
+import com.squareup.sqldelight.SqlDelightStatement;
+import java.lang.Long;
+import java.lang.Override;
+import java.lang.String;
+import java.util.Collections;
+
+public interface UserModel {
+  String TABLE_NAME = "users";
+
+  String ID = "id";
+
+  String AGE = "age";
+
+  String GENDER = "gender";
+
+  String CREATE_TABLE = ""
+      + "CREATE TABLE users (\n"
+      + "  id INTEGER PRIMARY KEY NOT NULL,\n"
+      + "  age INTEGER NOT NULL DEFAULT 0,\n"
+      + "  gender TEXT NOT NULL\n"
+      + ")";
+
+  long id();
+
+  int age();
+
+  @NonNull
+  String gender();
+
+  interface SelectWithFunctionsModel {
+    @NonNull
+    String gender();
+
+    @NonNull
+    String date();
+
+    @NonNull
+    String time();
+
+    @NonNull
+    String datetime();
+
+    @NonNull
+    String strftime();
+
+    @NonNull
+    String char_();
+
+    @NonNull
+    String hex();
+
+    @NonNull
+    String quote();
+
+    @NonNull
+    String soundex();
+
+    @NonNull
+    String sqlite_compileoption_get();
+
+    @NonNull
+    String sqlite_source_id();
+
+    @NonNull
+    String sqlite_version();
+
+    @NonNull
+    String typeof();
+
+    @NonNull
+    String lower_age();
+
+    @NonNull
+    String ltrim_gender();
+
+    @NonNull
+    String printf();
+
+    @NonNull
+    String replace();
+
+    @NonNull
+    String rtrim_gender();
+
+    @NonNull
+    String substr();
+
+    @NonNull
+    String trim_gender();
+
+    @NonNull
+    String upper_gender();
+
+    @NonNull
+    String group_concat_age();
+
+    long changes();
+
+    long last_insert_rowid();
+
+    long random();
+
+    long sqlite_compileoption_used();
+
+    long total_changes();
+
+    long count();
+
+    long instr();
+
+    long length_gender();
+
+    long unicode_gender();
+
+    @NonNull
+    byte[] randomblob();
+
+    @NonNull
+    byte[] zeroblob();
+
+    double total_age();
+
+    double avg_age();
```

Comment:
Just noticed that this is actually not 100% correct. According [to the docs](https://www.sqlite.org/lang_aggfunc.html#avg) `avg()` can be null if all arguments are null (e.g. `select avg(null)` will yield `null`). Then again, it's a corner case and other functions like `sum`, `min` etc. are also nullable under the same conditions and SQLDelight emits primitives for those too. I guess boxing the values isn't worth it?